staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go

```diff
@@ -128,3 +132,31 @@ func (s *SelectionPredicate) MatchesSingle() (string, bool) {
 func (s *SelectionPredicate) Empty() bool {
 	return s.Label.Empty() && s.Field.Empty()
 }
+
+// For any index defined by IndexFields, if a matcher can match only (a subset)
+// of objects that return <value> for a given index, a pair (<index name>, <value>)
+// wil be returned.
+func (s *SelectionPredicate) MatcherIndex() []MatchValue {
+	var result []MatchValue
+	for _, field := range s.IndexFields {
+		if value, ok := s.Field.RequiresExactMatch(strings.TrimPrefix(field, "f:")); ok {
+			result = append(result, MatchValue{IndexName: field, Value: value})
```

Comment:
Like, you could prepend the "f:" here if you wanted?

Reply:
It seems better. fix it.